src/algorithms/local_search/local_search.cpp

```diff
@@ -758,6 +772,16 @@ void LocalSearch<Route,
             continue;
           }
 
+          const auto& s_v = _input.vehicles[s_t.first];
+          const auto& t_v = _input.vehicles[s_t.second];
+
+          if (s_rank + _sol[s_t.second].size() >
+                t_rank + s_v.max_number_of_tasks or
+              t_rank + _sol[s_t.first].size() >
+                s_rank + t_v.max_number_of_tasks) {
```

Comment:
My bad. This was actually not a good idea since `t_rank + s_v.max_number_of_tasks` will overflow whenever number of tasks is `std::numeric_limits<size_t>::max()`. Maybe we could keep the previous version but reorder and group computation for values that we know are positive: `(_sol[s_t.second].size() - t_rank)` and `(_sol[s_t.first].size() - s_rank)`.